core/interfaces/i_accessibility.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview AST Node and navigation interfaces.
+ * @author samelh@google.com (Sam El-Husseini)
+ */
+
+'use strict';
+
+goog.provide('Blockly.IASTNodeLocation');
+goog.provide('Blockly.IASTNodeLocationWithBlock');
+goog.provide('Blockly.IBlocklyActionable');
+
+/**
+ * An AST node location interface.
+ * @interface
+ */
+Blockly.IASTNodeLocation = function() {};
+
+
+/**
+ * An AST node location that has an associated block.
+ * @interface
+ * @extends {Blockly.IASTNodeLocation}
+ */
+Blockly.IASTNodeLocationWithBlock = function() {};
```

Comment:
Should this be specific to ASTNodes? Or should it just be an interface for something that has a source block?

Reply:
It's possible there might be a need for an interface that only implements `getSourceBlock`, but for the use case that ASTNode is relying on this interface on is to assert an ASTNodeLocation that also has `getSourceBlock`. If we later have a need or a `getSourceBlock` interface then we can extend from that.